core/src/rpc/args.rs

```diff
@@ -0,0 +1,62 @@
+use crate::sql::Array;
+use crate::sql::Value;
+
+use super::rpc_error::RpcError;
+
+pub trait Take {
+	fn needs_one(self) -> Result<Value, RpcError>;
+	fn needs_two(self) -> Result<(Value, Value), RpcError>;
+	fn needs_one_or_two(self) -> Result<(Value, Value), RpcError>;
+	fn needs_one_two_or_three(self) -> Result<(Value, Value, Value), RpcError>;
+}
+
+impl Take for Array {
+	/// Convert the array to one argument
+	fn needs_one(self) -> Result<Value, RpcError> {
+		if self.len() != 1 {
+			return Err(RpcError::InvalidParams);
+		}
+		let mut x = self.into_iter();
+		match x.next() {
+			Some(a) => Ok(a),
+			None => Ok(Value::None),
+		}
+	}
+	/// Convert the array to two arguments
+	fn needs_two(self) -> Result<(Value, Value), RpcError> {
+		if self.len() != 2 {
+			return Err(RpcError::InvalidParams);
+		}
+		let mut x = self.into_iter();
+		match (x.next(), x.next()) {
+			(Some(a), Some(b)) => Ok((a, b)),
+			(Some(a), None) => Ok((a, Value::None)),
+			(_, _) => Ok((Value::None, Value::None)),
```

Comment:
Do we want to fix these in this PR or separately? We need to have a look at these methods again to optimize them, the last two cases should both return an error and be a catchall, as they will never trigger (see len check above)

Reply:
yes, a separate pr. There's lots of other refactoring and improvements to be done, but its not blocking. We can get that in for 1.5